sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/util/AvroUtils.java

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.internal.avro.implementation.util;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Utility methods for Avro.
+ */
+public class AvroUtils {
+
+    /**
+     * Converts a List of ByteBuffers into a byte array.
+     * @param bytes The buffers to convert.
+     * @return The byte array.
+     */
+    public static byte[] getBytes(List<?> bytes) {
+        long longTotalBytes = bytes
+            .stream()
+            .mapToLong(buffer -> {
+                checkByteBuffer("'buffer'", buffer);
+                return ((ByteBuffer) buffer).remaining(); })
+            .sum();
+
+        if (longTotalBytes > Integer.MAX_VALUE) {
+            throw new IllegalArgumentException("Bytes can not fit into a single array.");
+        }
+
+        int totalBytes = Math.toIntExact(longTotalBytes);
+
+        byte[] ret = new byte[totalBytes];
+        AtomicInteger offset = new AtomicInteger();
+        bytes.forEach(buffer -> {
+            checkByteBuffer("'buffer'", buffer);
+            ByteBuffer b = (ByteBuffer) buffer;
+            int length = b.remaining();
+            ((ByteBuffer) buffer).get(ret, 0, length);
+            offset.addAndGet(length);
+        });
+
+        return ret;
+    }
+
+    /**
+     * Checks whether an Object is of type Long.
+     * @param name The name of the variable
+     * @param variable The variable.
+     */
+    public static void checkLong(String name, Object variable) {
```

Comment:
Instead of defining method per each type you could pass `Class type` and use `isAssignableFrom` instead of `instanceof` .

See https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#isAssignableFrom-java.lang.Class-

Reply:
ooh that's really nice. I was wondering how I could make a helper method to do this

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/10463